src/main/java/com/conveyal/datatools/manager/controllers/api/DeploymentController.java

```diff
@@ -482,92 +482,48 @@ private static String deploy (Request req, Response res) {
-    private static String uploadToS3 (Request req, Response res) throws IOException {
+    private static Deployment uploadToS3 (Request req, Response res) {
         // Check parameters supplied in request for validity.
-        Auth0UserProfile userProfile = req.attribute("user");
         Deployment deployment = getDeploymentWithPermissions(req, res);
 
         String url;
-        Exception failure = null;
-
-        // Get file from request
-        if (req.raw().getAttribute("org.eclipse.jetty.multipartConfig") == null) {
-            MultipartConfigElement multipartConfigElement = new MultipartConfigElement(System.getProperty("java.io.tmpdir"));
-            req.raw().setAttribute("org.eclipse.jetty.multipartConfig", multipartConfigElement);
-        }
-        String extension = null;
-        File tempFile = null;
         try {
-            Part part = req.raw().getPart("file");
-            extension = "." + part.getContentType().split("/", 0)[1];
-            tempFile = File.createTempFile(part.getName() + "_csv_upload", extension);
-            InputStream inputStream;
-            inputStream = part.getInputStream();
-            FileOutputStream out = new FileOutputStream(tempFile);
-            IOUtils.copy(inputStream, out);
 
             String keyName = String.join(
                     "/",
                     bundlePrefix,
                     deployment.projectId,
                     deployment.id,
                     // Where filenames are generated. Prepend random UUID to prevent overwriting
-                    UUID.randomUUID() + "_" + part.getSubmittedFileName()
+                    UUID.randomUUID() + ""
             );
-            url = S3Utils.getDefaultBucketUrlForKey(keyName);
-            S3Utils.getDefaultS3Client().putObject(new PutObjectRequest(
-                    S3Utils.DEFAULT_BUCKET, keyName, tempFile)
-                    // Allow public read
-                    // TODO: restrict?
-                    .withCannedAcl(CannedAccessControlList.PublicRead));
-            return url;
-        } catch (IOException | ServletException | AmazonServiceException | CheckedAWSException e) {
+            url = SparkUtils.uploadMultipartRequestBodyToS3(req, "csvUpload", keyName);
+
+            // Update deployment csvs
+            List<String> updatedCsvList = new ArrayList<>(deployment.peliasCsvFiles);
+            updatedCsvList.add(url);
+
+            // If this is set, a file is being replaced
+            String s3FileToRemove = req.raw().getHeader("urlToDelete");
+            if (s3FileToRemove != null) {
+                updatedCsvList.remove(s3FileToRemove);
+            }
+
+            // Persist changes after removing deleted csv files from s3
+            removeDeletedCsvFiles(updatedCsvList, deployment, req);
+            return Persistence.deployments.updateField(deployment.id, "peliasCsvFiles", updatedCsvList);
+
+        } catch (AmazonServiceException e) {
             e.printStackTrace();
```

Comment:
If we fail with an exception, we should call logMessageAndHalt.

Reply:
Good catch thank you! Added in 4b73157ca781655923e944dad2ed1ccb0c32ce05